### ISSUE TITLE
NScurl::http GET link does not accept special characters (e.g. '+', '(', ')' etc)

### ISSUE THREAD

**Issue by dianapatru** [negrutiu/nsis-nscurl#13]
Hello,

I am using this plugin for an installer and I encountered a case when the NScurl::http GET does not work if the link contains special characters in the link (e.g. '+', '(', ')' etc). I tried to escape or encode before these characters but I received the same error.

This is how I tested the code:

1. `NScurl::http GET "https://github.com/dianapatru/example/blob/main/file_+_1.txt"  ".\output_file" /INSIST /CONNECTTIMEOUT 15s /CANCEL /RESUME /END`
`Pop $R0 ; Get the return value `

2. `NScurl::http GET "https://github.com/dianapatru/example/blob/main/file_%2B_1.txt"  ".\output_file" /INSIST /CONNECTTIMEOUT 15s /CANCEL /RESUME /END`
`Pop $R0 ;Get the return value`

And this is the value of R0: `HTTP server doesn't seem to support byte ranges. Cannot resume.`

If I use another plugin `inetc::get` with the same link, the return value is OK

Can you please help me on how can I fix this issue? 
Also, thank you for developing this plugin.

**Comment by negrutiu** [comment]
Hi Diana, thanks for reporting
My guess is that it's the `/RESUME` parameter causing all the trouble.
Please try this and see how it goes:
```{nsis}
NScurl::http GET "https://raw.githubusercontent.com/dianapatru/example/main/file_+_1.txt"  ".\output_file" /INSIST /CONNECTTIMEOUT 15s /CANCEL /END
```
What's different:
- Changed the URL to `https://raw.githubusercontent.com/...` to download the actual file. The original URL would download the github webpage displaying file_+_1.txt
- Removed `/RESUME` parameter

**Comment by negrutiu** [comment]
...not yet sure if `/RESUME` is buggy on my side or disabled on the GitHub server...
Need more time to investigate

**Comment by dianapatru** [comment]
Hello, 

Thank you for getting back to me. The option to remove the `/RESUME` parameter worked and now the behavior is as expected. 
The desired approach would be to still use the `/RESUME` option in the future. I would check from time to time to see if a fix was found for this issue. 

Thank you again for your help!

**Comment by negrutiu** [comment]
Issue fixed in [nscurl/1.2024.4.30](https://github.com/negrutiu/nsis-nscurl/releases/tag/v1.2024.4.30)
Internal transfer compression (aka the `Accept-Encoding` http header) was interfering with resuming.
Compression is now deactivated by default. A new parameter `/ENCODING` was added to enable it on-demand.
Happy installing! ;-)
